### PR TITLE
[SWE-28] Edit Profile Page Frontend

### PR DESCRIPTION
**Pull request by sammyoto**
Created a basic frontend for editing a profile page. I apologize for how ugly it is. Also @datderek I basically just copied all of your code from the post form Modal. Thanks.

### CODE REVIEW

File: app/dashboard/edit-profile/page.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { redirect } from "next/navigation";
+import styles from "../../../styles/Dashboard.module.css";
+import EditProfileForm from "../../components/EditProfileForm";
+
+export const metadata = {
+  title: 'EditProfile'
+}
+
+export default async function EditProfile() {
+  // Packages cookies into request header
```
Comment (jackfales) [inline app/dashboard/edit-profile/page.tsx:12]:
```suggestion
```

**Review by jackfales — APPROVED** [review]

File: app/components/EditProfileForm.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use client'
+import { FormEvent } from 'react';
+
+export default function EditProfileForm() {
+
+  /**
+   * Submits new post data then closes the modal
```
Comment (datderek) [inline app/components/EditProfileForm.tsx:7]: Change comment, not a modal.

**Review by datderek — APPROVED** [review]
Just have the `Edit Profile` button on the `/[user]` route redirect to the `/[user]/edit-profile`

**Comment by sammyoto** [comment]
Made a utility functions file for miscellaneous functions that might not need a whole file to themselves. Needed it so we can make client components for our pages.

**Comment by sammyoto** [comment]
Also good to know apparently you can't use redirect on client components where an event is involved https://nextjs.org/docs/app/api-reference/functions/redirect. Had to use the router for this one.

File: app/components/ClientUtilityFunctions.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * 
+ * @param isUser - true if current session user matches page, false otherwise
+ */
+export function EditProfileAndFollowButton({ user, isUser }: { user: string, isUser: boolean }) {
+    const { push } = useRouter();
+
+    const handleClick = (e) => {
+        e.preventDefault();
+        push(`${user}/edit-profile`);
+    }
```
Comment (datderek) [inline app/components/ClientUtilityFunctions.tsx:16]: Is it possible to use relative URLs with `router.push()`? If so, you wouldn't have to pass the `user` prop into the button.

```suggestion
export function EditProfileAndFollowButton(isUser: boolean) {
    const { push } = useRouter();

    const handleClick = (e) => {
        e.preventDefault();
        push('/edit-profile');
    }
```

**Review by datderek — APPROVED** [review]
LGTM. Just the relative URL comment if possible, then you should be good.

**Pull request merged**